infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -578,15 +581,18 @@ private void onStandbyClusterAddRemove(TopologyDescriptor discoveredTopology) {
         log.debug("Standby Cluster has been added or removed from topology={}", discoveredTopology);
 
         // We only need to process new standby's if your role is of an ACTIVE cluster
-        if (localClusterDescriptor.getRole() == ClusterRole.STANDBY) {
-            return;
-        }
-
-        if (replicationManager != null && isLeader.get()) {
-            replicationManager.processStandbyChange(discoveredTopology);
+        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE) {
+            if (replicationManager != null && isLeader.get()) {
+                replicationManager.processStandbyChange(discoveredTopology);
+            }
```

Comment:
If the topologyConfigId has been change, it will overrule the processStandbyChange. processStandbyChange only is called when the topologyConfigId doesn't change and the list of standbys have been changed. 
Is the logic correct that processStandbyChange called before updating the localTopology?

Reply:
Since we don't assume that config id change implies a role change anymore, processStandbyChange won't be overruled. Now we only detect if local node role changes or not, then perform `onClusterRoleChange`  or `onStandbyClusterAddRemove`



> Is the logic correct that processStandbyChange called before updating the localTopology?

I think the logic is correct, since stopping old replication relies on old config.